app.py

```diff
@@ -0,0 +1,634 @@
+)
+def disp_initial_sched(*vals):
+    if vals:
+        num_employees = vals[0] - 1
+        rand_seed = vals[1]
+    else:
+        num_employees = 9  # one less to account for trainee
+
+    now = datetime.now()
+    month = now.month
+    year = now.year
+    num_days = calendar.monthrange(year, month)[1]
+
+    shifts = [str(i + 1) for i in range(num_days)]
+    df = utils.build_random_sched(num_employees, shifts, rand_seed)
+
+    df.replace(False, " ", inplace=True)
+    df.replace(True, "X", inplace=True)
+
+    return utils.display_availability(df, month, year)
+
+
+# Don't allow max shifts to be smaller than min shifts
+@app.callback(Output("max shifts", "min"), Input("min shifts", "value"))
+def set_min_shifts(min_s):
+    return min_s
+
+
+# Don't allow min shifts to be bigger than max shifts
+@app.callback(Output("shifts min", "max"), Input("shifts max", "value"))
+def set_max_shifts(max_s):
+    return max_s
+
+
+# Don't allow max employees to be smaller than min employees
+@app.callback(Output("shifts max", "min"), Input("shifts min", "value"))
+def set_shifts_min(min_e):
+    return min_e
+
+
+# Don't allow min employees to be bigger than max employees
+@app.callback(Output("min shifts", "max"), Input("max shifts", "value"))
+def set_shifts_max(max_e):
+    return max_e
+
+
+@app.callback(
+    Output("built-sched", "children"),
+    Output("error_card_body", "children"),
+    Output("btn_solve_cqm", "n_clicks"),
+    Output("tabs", "active_tab"),
+    [
+        Input("btn_solve_cqm", "n_clicks"),
+        Input("checklist-input", "value"),
+        Input("min shifts", "value"),  # shifts per employee
+        Input("max shifts", "value"),
+        Input("shifts min", "value"),  # employees per shift
+        Input("shifts max", "value"),
+        Input("cons shifts", "value"),  # consecutive shifts allowed
+    ],
+    State("initial-sched", "children"),
+    State("built-sched", "children"),
+    State("error_card_body", "children"),
+)
+def on_button_click(
+    n,
+    checklist_value,
+    min_shifts,
+    max_shifts,
+    shift_min,
+    shift_max,
+    k,
+    sched_df,
+    built_sched,
+    e_card,
+):
+    if n is None:
+        return (
+            html.Label("Not constructed yet.", style={"max-width": "50%"}),
+            " ",
+            None,
+            "avail",
+        )
+    elif n == 0:
+        return built_sched, e_card, 0, "sched"
+    else:
+        shifts = list(sched_df["props"]["data"][0].keys())
+        shifts.remove("Employee")
+        availability = utils.availability_to_dict(
+            sched_df["props"]["data"], shifts
+        )
+        employees = list(availability.keys())
+
+        if 1 in checklist_value:
+            isolated = True
+        else:
+            isolated = False
+
+        if 2 in checklist_value:
+            manager = True
+        else:
+            manager = False
+
+        print("\nBuilding CQM...\n")
```

Comment:
I could use some help here - I tried to replicate what you did in your tour planning demo but it got complex pretty quick and I abandoned that approach. Is this something that you could help me with?

Reply:
Sure. I remember spending several days to get the submission with its time and colors working but what I don't remember is how modular it might or might not be. A much quicker and simpler alternative to start with can be adding a simple text window under the Solve CQM button to print out the submission messages. 

What is your schedule for working on this like? Would you like me to clone the example and make a PR to your branch at some point to add such an indication? Let me know what works best for you and when I can be of help.

Reply:
I'm out most of tomorrow and Monday, but plan to work through the rest of these on Tuesday. I think the goal was to get these done before 23 January. If you can do the PR before Tuesday that would be fantastic, but I know that's a short timeline for you.

Reply:
Okay, I can try getting you something to start with by EOD Monday so it won't conflict with you making changes. Once your updates are mostly done, we can improve it as needed.

Reply:
Now that this update of the example is not scheduled for the week after next, I've been pulled off to other work so not fully tested (though seems about right) but here are two PRs:

* [PR that adds indicator and uses tabs for advanced options](https://github.com/vgoliber/employee-scheduling-1/pull/2). I recommend merging just this one and then updating to your liking
* [PR that adds just the indicator](https://github.com/vgoliber/employee-scheduling-1/pull/1). I recommend declining in favor of the other PR